spswarehouse/powerschool/powerschool.py

```diff
@@ -0,0 +1,437 @@
+    elem.click()
+
+def powerschool_report_helper_type_in_element_by_id(driver: WebDriver, element_id: str, input_to_type: str):
+    """
+    Waits for an element by ID, clears it, and types in the input.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, element_id)))
+    elem.clear()
+    elem.send_keys(input_to_type)
+
+def powerschool_report_helper_type_in_element_by_name(driver: WebDriver, element_name: str, input_to_type: str):
+    """
+    Waits for an element by name, clears it, and types in the input.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.NAME, element_name)))
+    elem.clear()
+    elem.send_keys(input_to_type)
+
+def powerschool_report_helper_select_visible_text_in_element_by_id(driver: WebDriver, element_id: str, text_to_select: str):
+    """
+    Waits for an element by ID and selects it by specified text.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, element_id)))
+    select = Select(elem)
+    select.select_by_visible_text(text_to_select)
+
+def powerschool_report_helper_select_visible_text_in_element_by_name(driver: WebDriver, element_name: str, text_to_select: str):
+    """
+    Waits for an element by name and selects it by specified text.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.NAME, element_name)))
+    select = Select(elem)
+    select.select_by_visible_text(text_to_select)
+
+def powerschool_report_helper_click_element_by_id(driver: WebDriver, element_id: str):
+    """
+    Waits for an element by ID and clicks it.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, element_id)))
+    elem.click()
+
+def wait_for_new_file_in_folder(folder_path, original_files, max_attempts=20000):
+    """
+    Waits until a new file shows up in a folder. Loops until that is true or max_attempts is reached.
+
+    Parameters:
+    folder_path: The folder being monitored.
+    original_files: The list of files originally in the folder, before the new one is added.
+    max_attmepts: Optional parameter that sets how many loops the function will do.
+
+    Returns:
+    n/a
+    """
+    file_added = False
+    attempts = 0
+    while True and attempts < max_attempts:
+        for root, folders, files in os.walk(folder_path):
+            # break 'for' loop if files found
+            if len(files) > len(original_files):
+                file_added = True
+                break
+            else:
+                continue
+        # break 'while' loop if files found
+        if file_added:
+            # wait for download to complete fully after it's been added - hopefully 3 seconds is enough.
+            time.sleep(3)
+            break
+        attempts +=1
+
+def rename_recent_file_in_dir(folder, append_text):
```

Comment:
Also seems like an internal function?

Reply:
If you do keep this as an exposed function, I'd rename it.  The function is not exactly a rename; it's an append.